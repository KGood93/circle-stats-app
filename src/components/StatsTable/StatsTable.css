.meetupStats {
    margin: auto;
    margin-bottom: 20px;
    border: 2px solid rgb(12, 70, 94);
}

.tableHeader {
    font-family: 'Roboto', sans-serif;
    font-weight: bold;
    font-size: 24px;
    color: rgb(12, 70, 94);
    margin-bottom: 10px;
}

table {
    border-collapse: collapse;
}

table, th, td {
    font-family: 'Roboto', sans-serif;
    border: 1px solid rgb(12, 70, 94);
    padding: 3px;
}

th {
    border: 1px solid rgb(230, 248, 255);
    background-color: rgb(12, 70, 94);
    color: rgb(230, 248, 255);
}

.largeForm {
    display: none;
}

@media screen and (min-width: 450px) {
    .largeForm{
        display: inline-block;
    }
}