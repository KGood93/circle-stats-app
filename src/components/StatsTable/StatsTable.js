import React, {Component} from 'react'
import moment from 'moment'
import './StatsTable.css'

class StatsTable extends Component {

    renderTableHeader() {
        return (
            <tr className="columnHeader">
                <th>Date</th>
                <th>Location</th>
                <th>Attendees</th>
                <th className="largeForm">Notes</th>
            </tr>
        )
    }

    renderTableData() {
        return this.props.workingData.data.map((meetup, index) => {
            const {date, location, attendance, notes} = meetup
            return (
                <tr key={date}>
                    <td>{moment(date).format('YYYY-MM-DD')}</td>
                    <td>{location}</td>
                    <td>{attendance}</td>
                    <td className="largeForm">{notes}</td>
                </tr>
            )
        })
    }

    render() {
        return (
            <div className="StatsTable">
                <h3 className="tableHeader">Meetup Stats</h3>
                <table id='meetupStats'>
                    <tbody>
                        {this.renderTableHeader()}
                        {this.renderTableData()}
                    </tbody>
                </table>
            
            </div>
        )
    }
}

export default StatsTable