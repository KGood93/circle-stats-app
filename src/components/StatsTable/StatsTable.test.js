import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import StatsTable from './StatsTable';

describe('StatsTable Component', () => {
    it(`renders without crashing`, () => {
        const data = [
            {
                date: '2019-12-25',
                location: 'Parents House',
                attendance: 24,
                notes: 'Christmas',
            },
            {
                date: '2019-11-28',
                location: 'London',
                attendance: 12,
                notes: 'Thanksgiving',
            },
        ]
        const wrapper = shallow(<StatsTable workingData={data}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});