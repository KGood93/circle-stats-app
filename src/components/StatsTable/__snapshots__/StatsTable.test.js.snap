// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StatsTable Component renders without crashing 1`] = `
<div
  className="StatsTable"
>
  <h3
    className="tableHeader"
  >
    Meetup Stats
  </h3>
  <table
    className="meetupStats"
  >
    <tbody>
      <tr
        className="columnHeader"
      >
        <th
          className="allHead"
        >
          Date
        </th>
        <th
          className="allHead"
        >
          Location
        </th>
        <th
          className="allHead2"
        >
          Attendees
        </th>
        <th
          className="largeFormHead"
        >
          Notes
        </th>
      </tr>
      <tr
        className="dataBody"
        key="2019-12-25"
      >
        <td
          className="all"
        >
          2019-12-25
        </td>
        <td
          className="all"
        >
          Parents House
        </td>
        <td
          className="all"
        >
          24
        </td>
        <td
          className="largeForm"
        >
          Christmas
        </td>
      </tr>
      <tr
        className="dataBody"
        key="2019-11-28"
      >
        <td
          className="all"
        >
          2019-11-28
        </td>
        <td
          className="all"
        >
          London
        </td>
        <td
          className="all"
        >
          12
        </td>
        <td
          className="largeForm"
        >
          Thanksgiving
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
